tests/function_modifiers/test_macros.py

```diff
@@ -676,11 +679,103 @@ def test_pipe_output_end_to_end():
     assert result["chain_2_using_pipe_output"] == result["chain_2_not_using_pipe_output"]
 
 
-def test_pipe_output_end_to_end_2():
+# Mutate will mark the modules (and leave a mark).
+# Thus calling it a second time (for instance through pmultiple tests) might mess it up slightly...
+# Using fixtures just to be sure.
+
+
+@pytest.fixture()
+def _downstream_result_to_mutate():
+    def downstream_result_to_mutate() -> int:
+        return 2
+
+    yield downstream_result_to_mutate
+
+
+@pytest.fixture
+def import_mutate_module():
```

Comment:
Is the fixture at the right scope? For testing mutate we'll actually want it per-test... AS it modifies the function it decorates and leaves state.

Reply:
The pytest API says function is the default:  https://docs.pytest.org/en/6.2.x/reference.html#pytest.fixture

But doesn't hurt to make that explicit.

Reply:
Great, that works. And yeah, this is also for a future issue with x-module mutation